orchestrator/packages/common/src/logger.ts

```diff
@@ -1,26 +1,35 @@
 import pino from "pino";
 import { getConfig } from "./config";
 import path from "path";
+import { existsSync, mkdirSync } from "fs";
+
+const _CONFIG = getConfig();
+
+const pinoTarget = {
+  target: 'pino/file',
+  options: {
+    destination: _CONFIG.orchestratorLogsDir ?
+      path.join(_CONFIG.orchestratorLogsDir, `${_CONFIG.environment}.log`) :
+      1 // used by pino for STDOUT
+  }
+}
 
 const transport = pino.transport({
   targets: [
-    {
-      target: 'pino/file',
-      options: {
-        destination: path.join(__dirname, '../../../', 'logs', `${getConfig().environment}.log`),
-        mkdir: true
-      }
-    },
-    {
-      target: 'pino/file',
-      options: { destination: 2 }
-    }
+    pinoTarget
   ]
 });
 
-const logger = pino({
-  level: getConfig().environment === 'production' ? 'info' : 'debug',
-  timestamp: true,
-}, transport);
+const getLogger = () => {
+  if (_CONFIG.orchestratorLogsDir && !existsSync(_CONFIG.orchestratorLogsDir)) {
+    mkdirSync(_CONFIG.orchestratorLogsDir);
```

Comment:
This isn't recursive, so it only works if the leaf directory is missing; if middle parts of the path don't exist, this will fail.

Reply:
Fixed